corehq/apps/locations/models.py

```diff
@@ -468,6 +468,39 @@ def get_descendants(self, include_self=False, **kwargs):
             where, **kwargs
         )
 
+    @staticmethod
+    def get_bulk_ancestors(domain, location_ids, **kwargs):
+        """
+        Returns the grouped ancestors for the location ids passed in the
+        dictionary of following pattern
+        {location_id_1: [self, parent, parent_of_parent,.,.,.,],
+        location_id_2: [self, parent, parent_of_parent,.,.,.,],
+
+        }
+        :param domain: domain for which locations is to be pulled out
+        :param location_ids: locations ids whose ancestors needs to be find
+        :param kwargs: extra parameters
+        :return: dict
+        """
+        where = Q(domain=domain, location_id__in=location_ids)
+        location_ancestors = SQLLocation.objects.get_ancestors(where, **kwargs)
```

Comment:
This only needs `location_id`, `pk`, `name`, and `parent_id` - you could switch this to a `values_list` to reduce the columns returned.